src/API/Site/Controllers/Ads/ReportsController.php

```diff
@@ -63,6 +64,8 @@ protected function get_programs_report_callback(): callable {
 				$ads  = $this->container->get( AdsReport::class );
 				$data = $ads->get_report_data( 'campaigns', $this->prepare_query_arguments( $request ) );
 				return $this->prepare_item_for_response( $data, $request );
+			} catch ( ExceptionWithResponseData $e ) {
+				return new Response( $e->get_response_data( true ), $e->getCode() ?: 400 );
```

Comment:
💅 Wondering if would be better to add the specific Exception data in a 'data' prop. 

That way we can be sure the response data props are not mixed with the `Exception` props. 

Following the previous example:

```
        'message' => 'Error retrieving campaign: not found',
	'data' => [		
            'errors'  => [ 'NOT_FOUND' => 'not found' ],
	    'id'      =>  '123',
          ]
```

Reply:
> I don't mind adding the response data in a separate "data" prop, if that makes it easier. But that's not done in any of the other cases where an ExceptionWithResponseData is returned to the UI. So if we standardize that, we might need to rework some more code.

This is very related to my comments regarding the naming of the props. Having them in a wrapper would isolate the context a bit . However, I think it would be much better to apply a specific naming instead of generics, that way we can avoid the warapping.

`id` -> `report_id`, `campaing_id`, `merchant_id`...etc
`args` -> `report_args`, `query_args` etc
`type` -> `report_type`